crates/macros/src/parsing.rs

```diff
@@ -133,44 +167,26 @@ mod tests {
 
     #[test]
     fn test_rename_magic() {
-        for &(magic, expected) in &[
-            ("__construct", "__construct"),
-            ("__destruct", "__destruct"),
-            ("__call", "__call"),
-            ("__call_static", "__callStatic"),
-            ("__get", "__get"),
-            ("__set", "__set"),
-            ("__isset", "__isset"),
-            ("__unset", "__unset"),
-            ("__sleep", "__sleep"),
-            ("__wakeup", "__wakeup"),
-            ("__serialize", "__serialize"),
-            ("__unserialize", "__unserialize"),
-            ("__to_string", "__toString"),
-            ("__invoke", "__invoke"),
-            ("__set_state", "__set_state"),
-            ("__clone", "__clone"),
-            ("__debug_info", "__debugInfo"),
-        ] {
-            assert_eq!(magic, RenameRule::None.rename(magic));
-            assert_eq!(expected, RenameRule::Camel.rename(magic));
-            assert_eq!(expected, RenameRule::Pascal.rename(magic));
-            assert_eq!(expected, RenameRule::Snake.rename(magic));
-            assert_eq!(expected, RenameRule::ScreamingSnakeCase.rename(magic));
+        for magic in MAGIC_METHOD {
+            assert_eq!(magic, magic.rename_method(&RenameRule::None));
+            assert_eq!(magic, magic.rename_method(&RenameRule::Camel));
+            assert_eq!(magic, magic.rename_method(&RenameRule::Pascal));
+            assert_eq!(magic, magic.rename_method(&RenameRule::Snake));
+            assert_eq!(magic, magic.rename_method(&RenameRule::ScreamingSnakeCase));
         }
     }
 
     #[test]
     fn test_rename_php_methods() {
         let &(original, camel, snake, pascal, screaming_snake) =
             &("get_name", "getName", "get_name", "GetName", "GET_NAME");
-        assert_eq!(original, RenameRule::None.rename(original));
-        assert_eq!(camel, RenameRule::Camel.rename(original));
-        assert_eq!(pascal, RenameRule::Pascal.rename(original));
-        assert_eq!(snake, RenameRule::Snake.rename(original));
+        assert_eq!(original, original.rename(&RenameRule::None));
+        assert_eq!(camel, original.rename(&RenameRule::Camel));
+        assert_eq!(pascal, original.rename(&RenameRule::Pascal));
+        assert_eq!(snake, original.rename(&RenameRule::Snake));
         assert_eq!(
             screaming_snake,
-            RenameRule::ScreamingSnakeCase.rename(original)
+            original.rename(&RenameRule::ScreamingSnakeCase)
```

Comment:
Shouldn't this be `rename_method`. At least the test is called `test_rename_php_methods`

Reply:
Also probably want to add test for normal rename.

Reply:
No problem.
One question how bad it will be, if i delete prefix test in tests?
test_rename_magic_method -> rename_magic_method

Reply:
Would be fine with me

Reply:
Sorry for the delay. Will have a look this evening.

Reply:
No problem)